TSOClient/tso.simantics/NetPlay/EODs/Handlers/VMEODTwoPersonJobObjectMazePlugin.cs

```diff
@@ -0,0 +1,502 @@
+            int blanksPerPool = (MAX_ROWS * MAX_COLUMNS - MAX_COLORS * 4) - 2;
+            blanksPerPool /= 8;
+            AllColorPools = new List<ColorPool>()
+            {
+                new ColorPool(VMEODTwoPersonJobObjectMazePluginCellColors.Blank, blanksPerPool),
+                new ColorPool(VMEODTwoPersonJobObjectMazePluginCellColors.Blank, blanksPerPool),
+                new ColorPool(VMEODTwoPersonJobObjectMazePluginCellColors.Blue, MAX_COLORS),
+                new ColorPool(VMEODTwoPersonJobObjectMazePluginCellColors.Blank, blanksPerPool),
+                new ColorPool(VMEODTwoPersonJobObjectMazePluginCellColors.Blank, blanksPerPool),
+                new ColorPool(VMEODTwoPersonJobObjectMazePluginCellColors.Green, MAX_COLORS),
+                new ColorPool(VMEODTwoPersonJobObjectMazePluginCellColors.Blank, blanksPerPool),
+                new ColorPool(VMEODTwoPersonJobObjectMazePluginCellColors.Blank, blanksPerPool),
+                new ColorPool(VMEODTwoPersonJobObjectMazePluginCellColors.Red, MAX_COLORS + 1),
+                new ColorPool(VMEODTwoPersonJobObjectMazePluginCellColors.Blank, blanksPerPool),
+                new ColorPool(VMEODTwoPersonJobObjectMazePluginCellColors.Blank, blanksPerPool),
+                new ColorPool(VMEODTwoPersonJobObjectMazePluginCellColors.Yellow, MAX_COLORS + 1),
+            };
+
+            if (CurrentMaze == null) // first run
+            {
+                CurrentMaze = AbstractMazeGenerator<TSOMazeData>.GetEmptyMaze(MAX_ROWS, MAX_COLUMNS);
+                CurrentMaze.OnMazeGenerated += MazeGeneratedHandler;
+                CurrentMaze.OnFinalProcessingCell += OnProcessedCellHandler;
+            }
+            // random origin from the 5 choices of enum: BuildFromOrigins
+            var origin = Rng.Next(0, (int)BuildFromOrigins.Dead_Center + 1);
+            // also resets previous mazes
+            CurrentMaze.BuildFromOrigin(origin); 
+        }
+        private void SendCharismaPlayerData(bool disableInput)
+        {
+            if (CharismaPlayer != null)
+            {
+                CharismaPlayerHasInit = true;
+                if (disableInput)
+                    SendCharismaPlayerCellData("TSOMaze_Init_Cell");
+                else
+                    SendCharismaPlayerCellData("TSOMaze_Update_Cell");
+            }
+        }
+        private void SendCharismaPlayerCellData(string eventName)
+        {
+            byte wallConfig = (byte)CharismaPlayerOrigin.CellData.Wall_Config;
+            byte color = (byte)CharismaPlayerOrigin.CellData.Color;
+            CharismaPlayer.Send(eventName, new byte[] { wallConfig, color });
+        }
+        private void SendLogicPlayerData()
+        {
+            if (LogicPlayer != null)
+            {
+                LogicPlayerHasInit = true;
+                
+                // get the wall configs for half of the cells: every other one in each row, but staggered starting at 0 or 1 moving down rows
```

Comment:
This is a really weird way of doing this - I would have sent only the top&left wall of each tile. I guess this compresses better.